### PR TITLE
Bacnet version

### PR DESCRIPTION
**Pull request by JoelRedstone**
PR from grafnu: https://github.com/grafnu/daq/pull/161
References issue: https://github.com/faucetsdn/daq/issues/132

### CODE REVIEW

File: docs/device_report.md [hunk 1]
```diff
@@ -11,9 +11,14 @@
 
 | Test             |                        |
 |------------------|------------------------|
+<<<<<<< HEAD
+| Test report date | 2019-06-20T12:31:51.814Z |
+| DAQ version      | 1.0.0 |
+=======
```
Comment (InBrewJ) [inline docs/device_report.md]: @JoelRedstone There should never be any `<<<< HEAD` strings in commited code - it means a merge conflict hasn't been resolved properly

**Review by InBrewJ — CHANGES_REQUESTED** [review]

**Comment by JoelRedstone** [comment]
@InBrewJ PTAL

**Comment by InBrewJ** [comment]
@JoelRedstone LGTM. I'll do a bit of functional testing of this today.

File: subset/bacnet/bacnetTests/src/main/java/VersionTest.java [hunk 2]
```diff
@@ -131,10 +133,10 @@ private void formatProperty(Map.Entry<PropertyIdentifier, Encodable> property) {
   private void generateReport(String deviceMacAddress) {
```
Comment (InBrewJ) [inline subset/bacnet/bacnetTests/src/main/java/VersionTest.java:135]: **DISCLAIMER: this is not really do with this PR, just an observation**

@JoelRedstone @ewRedstone @ktredstone @MaximusJ08

Just a style point - I've noticed that @JoelRedstone, in this test, has a set of classes (e.g. Report.java):

https://github.com/JoelRedstone/daq/blob/bacnet_version/subset/bacnet/bacnetTests/src/main/java/helper/Report.java#L15

...that deal with things that you're doing in every single test; things like writing reports, handling connections and checking input parameters etc. @ewRedstone you're doing a similar thing here: 

https://github.com/ewRedstone/daq/blob/switchvalidation/subset/switches/src/main/java/switchtest/SwitchInterrogator.java#L677

I'd imagine you're doing similar but very slightly different things elsewhere, @ktredstone.

To make your lives easier, you guys could potentially have a `helper` set of functions in the `subset/` folder itself to standardise all of these things that are to do with DAQ specfic stuff so you can focus on other, (more fun?) functional things?

**Comment by InBrewJ** [comment]
@pisuke @JoelRedstone 

Code LGTM, a few queries:

- Should this test print out the actual bacnet version anywhere? It doesn't seem to do anything of that sort but this could be expected behaviour
- To **fully** functionally test this, we'd need to plug in a real bacnet device to an EasyIO/etc controller, right? Otherwise we can only verify that the `skip` condition works as expected
- @JoelRedstone, I'm guessing this:
https://github.com/JoelRedstone/daq/blob/bacnet_version/subset/bacnet/bacnetTests/src/main/java/Main.java#L12 is a `switch` statement because there will be more bacnet tests in the future?

**Review by InBrewJ — APPROVED** [review]

**Comment by pisuke** [comment]
> * Should this test print out the actual bacnet version anywhere? It doesn't seem to do anything of that sort but this could be expected behaviour

Yes it should print it out both in the report output in the module output section and in the Notes column in the summary table.

**Comment by pisuke** [comment]
> * To **fully** functionally test this, we'd need to plug in a real bacnet device to an EasyIO/etc controller, right? Otherwise we can only verify that the `skip` condition works as expected

Correct, we should test it in the Arup DAQ lab or Redstone DAQ lab. We have configured BACnet devices in both places.

**Comment by pisuke** [comment]
@JoelRedstone can you please add the BACnet version output in both the Notes and the test output.
Adding the test name and test description in the module output would also be great.

**Review by pisuke — CHANGES_REQUESTED** [review]
@JoelRedstone can you please add the BACnet version output in both the Notes column and in the module output. Including the test name and description in the module output would also be great. Thank you!

**Comment by JoelRedstone** [comment]
@InBrewJ Just to clarify:
The test currently write the version informations in the appendix.txt
You can also see this in the activate.log after you run daq.

**Comment by ewRedstone** [comment]
> @JoelRedstone can you please add the BACnet version output in both the Notes column and in the module output. Including the test name and description in the module output would also be great. Thank you!

Francesco the project management has said we are not to implement any new features such as test name and description. As this re-work of code that is already completed, You cannot keep changing the goal posts. Karen, Kathy and Damian should be notified of your request.

This also does not stop you accepting this pull request as you always want to commit "small" amounts of code so this is contradictory.

Test name and description will be added to future development cycle after existing sprint already proposed has been completed.

Description and Test Name is a nice to have and technically the test name already exists as part of the framework

**Comment by pisuke** [comment]
I really thought this was a very small change to be honest, let me handle it.

File: subset/bacnet/bacnetTests/src/main/java/VersionTest.java (hunk 2 above)
Comment (pisuke) [reply subset/bacnet/bacnetTests/src/main/java/VersionTest.java:135]: Very good idea!

**Comment by ewRedstone** [comment]
> I really thought this was a very small change to be honest, let me handle it.

A small change within the DAQ framework is a massive impact on time.

Finding the description information 10 minutes
Implementing the description information in the code 30 minutes
Testing the code with cmd/run -s 6 minutes
Testing the code with testing/test_aux.sh 15 minutes
Copying the inst/reports/report_9a02571e8f01_2019-07-10T100509+0000.md to docs/device_report.md
Testing the code again  with testing/test_aux.sh 15 minutes
Copying the out/test_aux.out to testing/test_aux.out
Testing the code again  with testing/test_aux.sh 15 minutes
Commiting to GitHub
Testing the code again  with testing/test_aux.sh 15 minutes
Copying the out/test_aux.out to testing/test_aux.out
Commiting to GitHub
Waiting for Travis to parse integration testing 25 minutes
If fail the monolithic process starts again......

2 hours at best, plus 20 minutes for setup time when we re-open parked code under development

**Comment by ewRedstone** [comment]
There is nothing stopping this PR going ahead.  Future PR's can be made to advance information

**Review by pisuke — APPROVED** [review]
I've implemented the protocol version output in the summary column.

**Pull request merged**